### PR TITLE
1.18.2 support and some fixes.

### PR DESCRIPTION
**Pull request by bsommerfeld**
We would like to integrate your changes into pathetic with your approval @aematsubara

### CODE REVIEW

File: pathetic-nms/v1_18_R2/pom.xml
Comment (bsommerfeld) [inline pathetic-nms/v1_18_R2/pom.xml:8]: This needs to be increased to 2.4. Unfortunately, we have already upgraded the versions in our trunk, so that 2.3 can no longer be resolved.

**Comment by aematsubara** [comment]
Of course, go ahead!

**Review by olijeffers0n — DISMISSED** [review]
Thanks! Before I would merge this I would be interested in if it would be possible to add support for the versions which were previously supported. Thanks for this though!

**Comment by aematsubara** [comment]
> Thanks! Before I would merge this I would be interested in if it would be possible to add support for the versions which were previously supported. Thanks for this though!

Sure, I think I found a solution for 1.15 & 1.16; is there a reason why versions 1.13 and 1.14 do not have an NMS implementation? I can add them if necessary

The reason why I used NMS to get the BlockState: https://github.com/patheloper/pathetic/commit/b03d20b3ab87bb89f0041b7b4a6df61e73f9bb95#r139351123

**Comment by olijeffers0n** [comment]
> > Thanks! Before I would merge this I would be interested in if it would be possible to add support for the versions which were previously supported. Thanks for this though!
> 
> Sure, I think I found a solution for 1.15 & 1.16; is there a reason why versions 1.13 and 1.14 do not have an NMS implementation? I can add them if necessary
> 
> The reason why I used NMS to get the BlockState: [b03d20b#r139351123](https://github.com/patheloper/pathetic/commit/b03d20b3ab87bb89f0041b7b4a6df61e73f9bb95#r139351123)

They don’t have NMS versions because we simply only supported the major versions. If you are happy to implement them they will be approved 😄

**Review by olijeffers0n — APPROVED** [review]

**Comment by bsommerfeld** [comment]
Thanks for contributing tho!

**Pull request merged**
